# HG changeset 9672 patch
From: kaf24@firebug.cl.cam.ac.uk
# Node ID 5da82db00c6e6dfa8dd33809c8fab94376635ad3
# Parent  5976f3828e1865da95f7931637466ae353824b3e
Subject: Simply do not declare module_exit() handlers for netback/blkback,
rather than declaring the modules unsafe.
Signed-off-by: Keir Fraser <keir@xensource.com>
xen-unstable changeset:   9970:60f7b567bb2b00d9dcf6ed86847feba4f9462177
xen-unstable date:        Wed May 10 16:47:00 2006 +0100

Acked-By: Jan Beulich <jbeulich@novell.com>
References: 17xxxx

Index: head-2006-05-11/drivers/xen/blkback/blkback.c
===================================================================
--- head-2006-05-11.orig/drivers/xen/blkback/blkback.c	2006-05-12 09:24:22.000000000 +0200
+++ head-2006-05-11/drivers/xen/blkback/blkback.c	2006-05-12 09:24:55.000000000 +0200
@@ -570,19 +570,12 @@ static int __init blkif_init(void)
 		list_add_tail(&pending_reqs[i].free_list, &pending_free);
     
 	blkif_xenbus_init();
-	__unsafe(THIS_MODULE);
+
 	return 0;
 }
 
 module_init(blkif_init);
 
-static void blkif_exit(void)
-{
-	BUG();
-}
-
-module_exit(blkif_exit);
-
 MODULE_LICENSE("Dual BSD/GPL");
 
 /*
Index: head-2006-05-11/drivers/xen/netback/netback.c
===================================================================
--- head-2006-05-11.orig/drivers/xen/netback/netback.c	2006-05-12 09:24:22.000000000 +0200
+++ head-2006-05-11/drivers/xen/netback/netback.c	2006-05-12 09:24:55.000000000 +0200
@@ -842,18 +842,10 @@ static int __init netback_init(void)
 		&netif_be_dbg);
 #endif
 
-	__unsafe(THIS_MODULE);
-
 	return 0;
 }
 
-static void netback_cleanup(void)
-{
-	BUG();
-}
-
 module_init(netback_init);
-module_exit(netback_cleanup);
 
 MODULE_LICENSE("Dual BSD/GPL");
 
